IdentityCore/src/cache/mac/MSIDMacKeychainTokenCache.m

```diff
@@ -317,6 +317,13 @@ - (BOOL)saveAccount:(MSIDAccountCacheItem *)account
     NSMutableDictionary *query = [self primaryAccountAttributesForKey:key];
     NSMutableDictionary *update = [self secondaryAccountAttributesForKey:key];
     update[(id)kSecValueData] = jsonData;
+
+    if ([self isRecentItem:query])
```

Comment:
this generates additional read request before each write resulting in worse performance.

Reply:
I have reduced this overhead by calling SecItemAdd() first, then only if it's an update check the lastMod timestamp before calling SecItemUpdate().

Reply:
Updated the implementation as discussed off-line so we utilize a single keychain read to read the previous item as well as it's last-modified info.